mongoutil/uri.go

```diff
@@ -123,3 +131,35 @@ func toCamelCase(s string) string {
 	}
 	return buf.String()
 }
+
+// TimeTagDocument can be used to add a creation timestamp attribute `_tim` and
+// to add or update an `updated_at` timestamp when it is called with reference
+// to a single mongo document.
+// Cf. https://github.com/mailgun/flagman/blob/92ffbc4/flagman/model/mixin.py#L51-L71
+// Unlike in the above example's Python implementation, this func must be explicitly called
+// immediately after document creation and every subsequent update to provide the same utility.
+//
+// id should be the value of the `_id` field assigned by mongodb to every document by default
+// c must be the mongo document collection that contains the document identified by the id argument
+// postInit should be true when this is being called immediately after document creation
+func TimeTagDocument(ctx context.Context, id primitive.ObjectID, c *mongo.Collection, postInit bool) error {
+	ctx, cancel := context.WithTimeout(ctx, 10*clock.Second)
+	defer cancel()
+
+	var update primitive.M
+	now := bsonx.DateTime(clock.Now().UTC().UnixNano() / 1e6)
+	filter := bson.M{"_id": id}
+
+	if postInit {
+		update = bson.M{"$set": bson.M{"_tim": now, "updated_at": now}}
+	} else {
+		update = bson.M{"$set": bson.M{"updated_at": now}}
+	}
+	if _, err := c.FindOneAndUpdate(ctx, filter, update, nil).DecodeBytes(); err != nil {
+		if err == mongo.ErrNoDocuments {
```

Comment:
Since Go 1.13 the idiomatic way is to use `errors.Is` for this - https://go.dev/blog/go1.13-errors